test/benchmark/factorial.js

```diff
@@ -40,19 +46,69 @@ function betterFactorial (n) {
   return prod
 }
 
-console.log('factorial(5) = ' + bigFactorial(new BigNumber(5)))
-console.log('new factorial(5) = ' + betterFactorial(new BigNumber(5)))
+function initializeFactorial (n) {
+  n = n.toNumber()
+
+  let loop = Math.floor(n / 2)
+  const f = [loop]
+
+  if (n % 2 === 1) {
+    f[loop] = n
+  }
+
+  let i = 1
+  let s = loop
+
+  for (let inc = loop - 1; inc > 0; inc--) {
+    s += inc
+    f[i] = s
+    while (f[i] % 2 === 0) {
+      loop++
+      f[i] = Math.floor(f[i] / 2)
+    }
+    i++
+  }
```

Comment:
Stylistically, I think it is better to use `Array.prototype.push` than assigining to indexes beyond the current end of the array. Could we rework this loop to something like this?

```js
  let s = loop

  for (let inc = loop - 1; inc > 0; inc--) {
    s += inc
    let nameMe = s;
    while (nameMe % 2 === 0) {
      loop++
      nameMe /= nameMe
    }
    f.push(nameMe)
  }

```

I removed the`Math.floor` as we already check that `nameMe` is a multiple of two before dividing it by two. I was unsure about how to name `nameMe` (so I didn't) but we would want a more descriptive name there.

Just a thought of mine, what do you reckon?

Reply:
I can take a look.